src/test/java/teammates/test/cases/action/InstructorCourseRemindActionTest.java

```diff
@@ -171,39 +171,31 @@ public void testExecuteAndPostProcess() throws Exception {
                 Const.ParamsNames.INSTRUCTOR_EMAIL, invalidEmail
         };
         
-        try {
-            remindAction = getAction(addUserIdToParams(instructorId, submissionParams));
-            redirectResult = (RedirectResult) remindAction.executeAndPostProcess();
-            signalFailureToDetectException();
-        } catch (EntityNotFoundException e) {
-            ignoreExpectedException();
-        }
+        testExecuteAndPostProcess_assertEntityNotFoundException(instructorId, submissionParams);
         
         submissionParams = new String[]{
                 Const.ParamsNames.COURSE_ID, courseId,
                 Const.ParamsNames.STUDENT_EMAIL, invalidEmail
         };
         
-        try {
-            remindAction = getAction(addUserIdToParams(instructorId, submissionParams));
-            redirectResult = (RedirectResult) remindAction.executeAndPostProcess();
-            signalFailureToDetectException();
-        } catch (EntityNotFoundException e) {
-            ignoreExpectedException();
-        }
+        testExecuteAndPostProcess_assertEntityNotFoundException(instructorId, submissionParams);
         
         submissionParams = new String[]{
                 Const.ParamsNames.COURSE_ID, "invalidCourseId"
         };
         
+        testExecuteAndPostProcess_assertEntityNotFoundException(instructorId, submissionParams);
+    }
+
+    private void testExecuteAndPostProcess_assertEntityNotFoundException(String instructorId,
+                                                                         String[] submissionParams) {
         try {
-            remindAction = getAction(addUserIdToParams(instructorId, submissionParams));
-            redirectResult = (RedirectResult) remindAction.executeAndPostProcess();
+            Action remindAction = getAction(addUserIdToParams(instructorId, submissionParams));
+            remindAction.executeAndPostProcess();
             signalFailureToDetectException();
```

Comment:
You may need to pass the name of expected exception to `signalFailureToDetectException();` to indicate which exception is expected.